third_party/terraform/resources/resource_bigquery_table.go

```diff
@@ -24,53 +22,113 @@ func checkNameExists(jsonList []interface{}) error {
 	return nil
 }
 
-// JSONBytesEqual compares the JSON in two byte slices.
-// Reference: https://stackoverflow.com/questions/32408890/how-to-compare-two-json-requests
-func JSONBytesEqual(a, b []byte) (bool, error) {
-	var j, j2 interface{}
-	if err := json.Unmarshal(a, &j); err != nil {
-		return false, err
-	}
-	if j == nil {
-		return false, fmt.Errorf("The old schema value was nil")
-	}
-	jList := j.([]interface{})
-	if err := checkNameExists(jList); err != nil {
-		return false, err
-	}
-	sort.Slice(jList, func(i, k int) bool {
-		return jList[i].(map[string]interface{})["name"].(string) < jList[k].(map[string]interface{})["name"].(string)
-	})
-	if err := json.Unmarshal(b, &j2); err != nil {
-		return false, err
-	}
-	if j2 == nil {
-		return false, fmt.Errorf("The new schema value was nil")
-	}
-	j2List := j2.([]interface{})
-	if err := checkNameExists(j2List); err != nil {
-		return false, err
-	}
-	sort.Slice(j2List, func(i, k int) bool {
-		return j2List[i].(map[string]interface{})["name"].(string) < j2List[k].(map[string]interface{})["name"].(string)
-	})
-	return reflect.DeepEqual(j2List, jList), nil
+func jsonCompareWithMapKeyOverride(a, b interface{}, compareMapKeyVal func(key string, val1, val2 map[string]interface{}) bool) (bool, error) {
```

Comment:
Can you describe in a comment what the map key override does, and how it affects the JSON comparison?